File: gerencia_de_estoque/src/main/Item.java
package main;

public abstract class Item {

    private String nome;
    private int quantidade;
    private String categoria;
    private double valor;
    private int id;

    public Item(String nome, int quantidade, String categoria, double valor, int id) {
        this.nome = nome;
        this.quantidade = quantidade;
        this.categoria = categoria;
        this.valor = valor;
        this.id = id;
    }

    public void atualizarCaracteristicasBasicas(String newNome, int newQuantidade,
                                                String newCategoria, double newValor, int newId) {
        nome = newNome;
        quantidade = newQuantidade;
        categoria = newCategoria;
        valor = newValor;
        id = newId;
    }

    public abstract void restringir();

    public abstract void liberar();

    // TODO: substituir esse método por toString(), redundante
    public String toString() {
        return String.format("""
                        ---Produto----
                        ID: %d
                        Nome: %s
                        Categoria: %s
                        Qtd: %d
                        Valor: R$%.2f
                        """, id, nome,
                categoria, quantidade, valor);
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getQuantidade() {
        return quantidade;
    }

    public void setQuantidade(int quantidade) {
        this.quantidade = quantidade;
    }

    public String getCategoria() {
        return categoria;
    }

    public void setCategoria(String categoria) {
        this.categoria = categoria;
    }

    public double getValor() {
        return valor;
    }

    public void setValor(double valor) {
        this.valor = valor;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

}

File: estoque/backend/model/item.py
from abc import ABC  # , abstractmethod


class Item(ABC):
    def __init__(self, nome: str, categoria: str, quantidade: int, id: int, valor: float, restrito: bool = False):
        self._nome = nome
        self._categoria = categoria,
        self._quantidade = quantidade
        self._id = id
        self._valor = valor
        self._restrito = restrito

    def restringir(self) -> None:
        self._restrito = True

    def liberar(self) -> None:
        self._restrito = False

    def listar_caracteristicas_basicas(self) -> str:
        return self._nome + self._categoria + str(self._quantidade) + str(self._id) + str(self._valor) + str(self._restrito)
